WordPress/Classes/Networking/PlansRemote.swift

```diff
@@ -54,9 +56,173 @@ private func mapPlansResponse(response: AnyObject) throws -> (activePlan: Plan,
+                                let json = try NSJSONSerialization.JSONObjectWithData(response, options: [])
+                                return (try mapPlanFeaturesResponse(json), modificationDate)
+                            }
+                        }
+                    }
+                }
+                catch {}
+            }
+        }
+        
+        return nil
+    }
+    
+    private func cacheDateIsValid(date: NSDate) -> Bool {
+        return NSCalendar.currentCalendar().isDateInToday(date)
+    }
+    
+    private func fetchPlanFeatures(success: PlanFeatures -> Void, failure: ErrorType -> Void) {
+        let endpoint = "plans/features"
+        let path = pathForEndpoint(endpoint, withVersion: ServiceRemoteRESTApiVersion_1_2)
+        
+        api.GET(path,
+            parameters: nil,
+            success: {
+                [weak self] requestOperation, response in
+                self?.cacheResponseData(requestOperation.responseData)
+                do {
+                    try success(mapPlanFeaturesResponse(response))
+                } catch {
+                    failure(error)
+                }
+            }, failure: {
+                _, error in
+                failure(error)
+        })
+    }
+    
+    private var cacheFileURL: NSURL? {
+        guard let cacheDirectory = NSFileManager.defaultManager().URLsForDirectory(.CachesDirectory, inDomains: .UserDomainMask).first else { return nil }
+        
+        return cacheDirectory.URLByAppendingPathComponent(cacheFilename)
+    }
+    
+    private func cacheResponseData(responseData: NSData?) {
+        guard let responseData = responseData else { return }
+        guard let cacheFileURL = cacheFileURL else { return }
+        
+        responseData.writeToURL(cacheFileURL, atomically: true)
+    }
+}
+
+private func mapPlanFeaturesResponse(response: AnyObject) throws -> PlanFeatures {
+    guard let json = response as? [[String: AnyObject]] else {
+        throw PlansRemote.Error.DecodeError
+    }
+    
+    var features = [PlanID: [PlanFeature]]()
+    for featureDetails in json {
+        if let slug = featureDetails["product_slug"] as? String,
+            let title = featureDetails["title"] as? String,
+            var description = featureDetails["description"] as? String,
+            let iconName = featureDetails["icon"] as? String {
+                
+                if let planDetails = featureDetails["plans"] as? [String: AnyObject] {
+                    for (planID, planInfo) in planDetails {
+                        if let planID = Int(planID) {
+                            if features[planID] == nil {
+                                features[planID] = [PlanFeature]()
+                            }
+                            
+                            if let planInfo = planInfo as? [String: String],
+                                let planSpecificDescription = planInfo["description"] {
```

Comment:
I'd turn all these `if`s into `guard`s. First, it's less indentation and makes code more readable. But also, it ensures you handle the unexpected cases properly. With the current code, if any of the condition fails the feature is just ignored and no error is notified.